Update dependency `pulldown-cmark` to v0.12.0 for `mdman`

### Problem

Renovate bot found a newer version available for dependency `pulldown-cmark` in <https://github.com/rust-lang/cargo/pull/14623>. `pulldown-cmark` is used by `mdman` for templatizing and generating man pages.

However, the new version has some API changes that need to be fixed:


<details><summary>Build failure. Click to See details</summary>
<p>

```
error[E0532]: expected unit struct, unit variant or constant, found tuple variant `TagEnd::BlockQuote`
   --> crates/mdman/src/format/man.rs:266:25
    |
2[66](https://github.com/rust-lang/cargo/actions/runs/11116205209/job/30886065074?pr=14623#step:16:67) |                         TagEnd::BlockQuote => {
    |                         ^^^^^^^^^^^^^^^^^^ help: use the tuple variant pattern syntax instead: `TagEnd::BlockQuote(_)`
    |
   ::: /Users/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pulldown-cmark-0.12.1/src/lib.rs:256:5
    |
256 |     BlockQuote(Option<BlockQuoteKind>),
    |     ---------- `TagEnd::BlockQuote` defined here

error[E0532]: expected unit struct, unit variant or constant, found tuple variant `TagEnd::BlockQuote`
   --> crates/mdman/src/format/text.rs:234:21
    |
234 |                     TagEnd::BlockQuote => {
    |                     ^^^^^^^^^^^^^^^^^^ help: use the tuple variant pattern syntax instead: `TagEnd::BlockQuote(_)`
    |
   ::: /Users/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pulldown-cmark-0.12.1/src/lib.rs:256:5
    |
256 |     BlockQuote(Option<BlockQuoteKind>),
    |     ---------- `TagEnd::BlockQuote` defined here

For more information about this error, try `rustc --explain E0532`.

```

</p>
</details> 

### Steps

1. Clone `rust-lang/cargo`
2. Run `cargo build-man`
3. Fix the build failure.

### Possible Solution(s)

Look into what have been changed in `pulldown-cmark` and update accordingly.

### Version

```text
15fbd2f607d4defc87053b8b76bf5038f2483cf4
```


<!-- TRIAGEBOT_START -->

<!-- TRIAGEBOT_ASSIGN_START -->

<!-- TRIAGEBOT_ASSIGN_DATA_START$${"user":"dacianpascu06"}$$TRIAGEBOT_ASSIGN_DATA_END -->

<!-- TRIAGEBOT_ASSIGN_END -->
<!-- TRIAGEBOT_END -->

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -76,7 +76,7 @@ pathdiff = "0.2.1"
 percent-encoding = "2.3.1"
 pkg-config = "0.3.30"
 proptest = "1.5.0"
-pulldown-cmark = { version = "0.11.0", default-features = false, features = ["html"] }
+pulldown-cmark = { version = "0.12.0", default-features = false, features = ["html"] }
 rand = "0.8.5"
 regex = "1.10.5"
 rusqlite = { version = "0.32.0", features = ["bundled"] }

--- a/crates/mdman/src/format/man.rs
+++ b/crates/mdman/src/format/man.rs
@@ -140,7 +140,7 @@ impl<'e> ManRenderer<'e> {
                                 suppress_paragraph = true;
                             }
                         }
-                        Tag::BlockQuote(_kind) => {
+                        Tag::BlockQuote(..) => {
                             self.flush();
                             // .RS = move left margin over 3
                             // .ll = shrink line length
@@ -256,14 +256,18 @@ impl<'e> ManRenderer<'e> {
                         Tag::Image { .. } => {
                             bail!("images are not currently supported")
                         }
-                        Tag::HtmlBlock { .. } | Tag::MetadataBlock { .. } => {}
+                        Tag::HtmlBlock { .. }
+                        | Tag::MetadataBlock { .. }
+                        | Tag::DefinitionList
+                        | Tag::DefinitionListTitle
+                        | Tag::DefinitionListDefinition => {}
                     }
                 }
                 Event::End(tag_end) => {
                     match &tag_end {
                         TagEnd::Paragraph => self.flush(),
                         TagEnd::Heading(..) => {}
-                        TagEnd::BlockQuote => {
+                        TagEnd::BlockQuote(..) => {
                             self.flush();
                             // restore left margin, restore line length
                             self.output.push_str(".br\n.RE\n.ll\n");
@@ -317,7 +321,12 @@ impl<'e> ManRenderer<'e> {
                                 write!(self.output, "<{}>", escape(&dest_url)?)?;
                             }
                         }
-                        TagEnd::Image | TagEnd::HtmlBlock | TagEnd::MetadataBlock(..) => {}
+                        TagEnd::Image
+                        | TagEnd::HtmlBlock
+                        | TagEnd::MetadataBlock(..)
+                        | TagEnd::DefinitionListDefinition
+                        | TagEnd::DefinitionListTitle
+                        | TagEnd::DefinitionList => {}
                     }
                 }
                 Event::Text(t) => {

--- a/crates/mdman/src/format/text.rs
+++ b/crates/mdman/src/format/text.rs
@@ -222,7 +222,11 @@ impl<'e> TextRenderer<'e> {
                         Tag::Image { .. } => {
                             bail!("images are not currently supported")
                         }
-                        Tag::HtmlBlock { .. } | Tag::MetadataBlock { .. } => {}
+                        Tag::HtmlBlock { .. }
+                        | Tag::MetadataBlock { .. }
+                        | Tag::DefinitionList
+                        | Tag::DefinitionListTitle
+                        | Tag::DefinitionListDefinition => {}
                     }
                 }
                 Event::End(tag_end) => match &tag_end {
@@ -231,7 +235,7 @@ impl<'e> TextRenderer<'e> {
                         self.hard_break();
                     }
                     TagEnd::Heading(..) => {}
-                    TagEnd::BlockQuote => {
+                    TagEnd::BlockQuote(..) => {
                         self.indent -= 3;
                     }
                     TagEnd::CodeBlock => {
@@ -274,7 +278,12 @@ impl<'e> TextRenderer<'e> {
                             write!(self.word, "<{}>", dest_url)?;
                         }
                     }
-                    TagEnd::Image | TagEnd::HtmlBlock | TagEnd::MetadataBlock(..) => {}
+                    TagEnd::HtmlBlock { .. }
+                    | TagEnd::MetadataBlock { .. }
+                    | TagEnd::DefinitionList
+                    | TagEnd::DefinitionListTitle
+                    | TagEnd::Image
+                    | TagEnd::DefinitionListDefinition => {}
                 },
                 Event::Text(t) | Event::Code(t) => {
                     if wrap_text {

